restapi/src/main/java/io/stargate/web/resources/RowResource.java

```diff
@@ -314,17 +336,26 @@ public Response queryRows(
                   .build();
             }
 
-            orderByExpression = "ORDER BY " + name + " " + direction;
+            Column.Order colOrder = "ASC".equals(direction) ? Column.Order.Asc : Column.Order.Desc;
+            order.add(ImmutableColumnOrder.of(name, colOrder));
           }
 
-          String query =
-              String.format(
-                  "SELECT %s FROM %s.%s WHERE %s %s",
-                  returnColumns, keyspaceName, tableName, expression, orderByExpression);
-          CompletableFuture<ResultSet> selectQuery =
-              localDB.query(query.trim(), ConsistencyLevel.LOCAL_QUORUM, values.toArray());
+          ResultSet r =
+              db.getAuthorizationService()
+                  .authorizedDataRead(
+                      () ->
+                          localDB
+                              .query()
+                              .select()
+                              .column(columns)
+                              .from(tableMetadata.keyspace(), tableMetadata.name())
+                              .where(wheres)
+                              .orderBy(order)
+                              .consistencyLevel(ConsistencyLevel.LOCAL_QUORUM)
+                              .execute(),
+                      token,
+                      wheres.stream().map(TargetCell::new).collect(Collectors.toList()));
```

Comment:
Would it be possible to pass the statement object (up to the `.execute()` call) as a parameter and execute it inside `authorizedDataRead`? This way we won't have to process `wheres` in two places in callers. WDYT?

Reply:
Sorry, not sure I follow.

Reply:
I mean extracting `wheres` from the statement object. I realize that this will require some refactoring of `QueryBuilderImpl`, so I'm fine with deferring this until we know all the client-side use cases.